### ISSUE TITLE
Displayed numerical value of parameter with a displayUnit modifier is applied to it is wrong

### ISSUE THREAD

**Issue by ceraolo** [OpenModelica/OpenModelica#11846]
### Description
This is probably related to #11840

### Steps to Reproduce
This requires a [develop](https://github.com/PowerGrids/PowerGrids/tree/develop) version of PowerGrids dated 2024. 
Consider this model:

 ```modelica
model Test
  PowerGrids.Electrical.PowerFlow.BusPF bus33(UNom(displayUnit = "V") = 1) annotation(
    Placement(transformation(origin = {60, 10}, extent = {{-10, -10}, {10, 10}})));
equation
 end Test;
```

UNom should be 1V but it is erroneously displaied in the mask as 0.001 V. Furthermore it is impossible to set 1V through the mask.

### Version and OS
OM 1.23.0-dev214

**Comment by casella** [comment]
@adeas31, as I understand the problem here is that OMEdit uses the value of the `displayUnit` attribute declared in the modifier for bus33 (which is "V") to set the unit drop-down menu, but then uses instead the `displayUnit` attribute of the parameter as defined in the `BusPF` class (which is `"kV"`) to actually compute the displayed value.

The only way to get the right value to display is then to first change the unit from "V" to any other unit, then change it back to "V". Only selecting "V" doesn't have any effect, I guess because the callback of the drop-down menu is only called if the selection is changed.

This needs to be fixed with high priority.

**Comment by casella** [comment]
Thanks @ceraolo for reporting!

**Comment by adeas31** [comment]
I can't reproduce this issue on the master. I believe this is same as #11782 and is already fixed with #11825.

**Comment by casella** [comment]
I tested this when the nightly builds were stalled, so I did not have the version on master, sorry about that.
